test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -60,5 +60,98 @@ public void FirstExperienceWithML()
                 PetalLength = 5.1f
             });
         }
+
+
+        [Fact]
+        public void SimpleIrisDescisionTrees()
+        {
+            var env = new TlcEnvironment(new SysRandom(0), verbose: true);
+            string dataPath = "iris-data.txt";
+            // Create reader with specific schema.
+            var dataReader = TextLoader.CreateReader(env, ctx => (
+               Label: ctx.LoadText(0),
+               SepalWidth: ctx.LoadFloat(1),
+               SepalLength: ctx.LoadFloat(2),
+               PetalWidth: ctx.LoadFloat(3),
+               PetalLength: ctx.LoadFloat(4)),
+               dataPath);
+
+            // Load the data into the system.
+            var data = dataReader.Read(dataPath);
+
+            var preprocess = data.Schema.MakeEstimator(row => (
+                // Convert string label to key.
+                Label: row.Label.Dictionarize(),
+                // Concatenate all features into a vector.
+                Features: row.SepalWidth.ConcatWith(row.SepalLength, row.PetalWidth, row.PetalLength)));
+
+
+            var pipeline = preprocess
+                .Append(row => row.Label.PredictWithDecisionTrees(row.Features))
+                // shoul it be BagVectorize instead of KeyToValue?
+                .Append(row => row.PredictedLabel.KeyToValue());
+
+            var model = pipeline.Fit(data);
+
+            var predictions = model.Transform(dataReader.Read(testDataPath));
+            var evaluator = new MultiClassEvaluator(env);
+            var metrics = evaluator.Evaluate(predictions);
+        }
+
+        [Fact]
+        public void TwitterSentimentAnalysis()
+        {
+            var env = new TlcEnvironment(new SysRandom(0), verbose: true);
+            var dataPath = "wikipedia-detox-250-line-data.tsv";
+            // Load the data into the system.
+            var data = TextLoader.CreateReader(env, ctx => (
+                   Label: ctx.LoadFloat(0),
+                   Text: ctx.LoadText(1)),
+                   dataPath, hasHeader: true).Read(dataPath);
+
+            var preprocess = data.Schema.MakeEstimator(row => (
+                Label: row.Label,
+                // Concatenate all features into a vector.
+                Features: row.Text.TextFeaturizer()));
+
+            var pipeline = preprocess.
+                Append(row => row.Label.TrainLinearClassification(row.Features));
+
+            var (trainData, testData) = CrossValidator.TrainTestSplit(env, data: data, trainFraction: 0.7);
```

Comment:
I'm afraid I don't understand what is meant to be promoted here. Cross-validation suggests to me that there's some sort of loop, with independent featurization happening somewhere. And in cross-validation specifically, that helper function -- assuming we agree that it is necessary to have one -- would handle the splitting, according to the number of folds, not by some sort of `trainFraction` parameter.

Reply:
If the idea is just some sort of utility to split the dataset in two, we cannot name whatever that utility is, cannot have anything to do with cross-validation, since that already refers to a very specific way of splitting up data.

Reply:
Okay, okay, so not `CrossValidator` (I was just trying to follow the example in sklearn, but I'm not too fond of it). But still, the data instances will not be `DataView<TPig>`, they will be `IDataView`, unless we make `TrainTestSplit` work in pigsty-land, right? I suppose we should actually do this, so line 121 is possible

---
In reply to: [215061185](https://github.com/dotnet/machinelearning/pull/809#discussion_r215061185) [](ancestors = 215061185)

Reply:
Sorry, why wouldn't the instances be `DataView<TPig>`? So you feed in `DataView<TPig>`, and you get two (or even more, if we're ambitious) possibly stratified instances.

Reply:
That sounds good to me

---
In reply to: [215095002](https://github.com/dotnet/machinelearning/pull/809#discussion_r215095002) [](ancestors = 215095002)